### PR TITLE
CODEOWNERS cleanup

### LINKED ISSUE #1195: CODEOWNERS additions and cleanup
**Describe the feature request**

Add the @vmware-tanzu/tkg-infra-x  team to codeowners to create continuity and move away from single points of failure for reviews related to networking, windows, loadbalancing.

Along the way, also ensure there is a review team for docs, as well as clean up individual ids in the list when they are redundant.

**Describe alternatives you've considered**

Add individuals.

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Additional context**

### PR DESCRIPTION
**Pull request by benjaminapetersen**
Signed-off-by: Benjamin A. Petersen <ben@benjaminapetersen.me>

### What this PR does / why we need it

- Teams should maintain their docs under `/docs` as well as the code they own.  Adding my own team first.
- add infra-x team as owners on the test repo (largely depended on by windows and ipv6 work)
- sort by path

### Which issue(s) this PR fixes

Fixes #1195

### Release note

```release-note

```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Comment by ankeesler** [comment]
seems reasonable to me, altho i don’t think `@vmware-tanzu/tkg-iam-owners` should have permissions for the whole `/docs` tree bc there is a bunch of stuff in there that is not IAM related. i could see us having approval over specific CLI doc files? e.g., `docs/cli/commands/tanzu_pinniped-auth*`

**Comment by benjaminapetersen** [comment]
NIce, I agree. Updated.

**Comment by ankeesler** [comment]
cool. thanks ben. one unknown i have - are there docs folks that actually oversee these files? maybe a core reviewer could answer that question when they review this pr.

File: CODEOWNERS [hunk 1]
```diff
@@ -3,6 +3,7 @@
 /addons/pinniped @vmware-tanzu/tanzu-framework-reviewers @vmware-tanzu/tkg-iam-owners
 /cmd/cli @vmware-tanzu/tanzu-framework-reviewers @Anuj2512
 /cmd/cli/plugin/tkr @vmware-tanzu/tanzu-framework-reviewers @prkalle
+/docs/cli/commands/tanzu_pinniped-auth*  @vmware-tanzu/tkg-iam-owners
```
Comment (stmcginnis) [inline CODEOWNERS:6]: Can you change the owners to include the framework reviewers too?

```
@vmware-tanzu/tanzu-framework-reviewers @vmware-tanzu/tkg-iam-owners
```

**Comment by vuil** [comment]
> seems reasonable to me, altho i don’t think `@vmware-tanzu/tkg-iam-owners` should have permissions for the whole `/docs` tree bc there is a bunch of stuff in there that is not IAM related. i could see us having approval over specific CLI doc files? e.g., `docs/cli/commands/tanzu_pinniped-auth*`

I see the PR is updated to pattern match a particular group of cli command generated doc files. But in general there are going to be a bunch of files/dirs in docs/ that are better owned by a specific team and how to map those without causing a sprawl in CODEOWNERS file is what I am thinking about. I can see other teams would have required multiple lines in the CODEOWNERS to achieve the equivalent mapping.

Was there an earlier version of the change that directly maps /docs to a team?
For the time being how about going back to that suggestion but just add all the reviewer teams to it.
The downside of have more teams being able to approve changes to docs/ to is fairly low imo.

**Comment by benjaminapetersen** [comment]
@vuil this was the previous version:

```
/docs* @vmware-tanzu/tanzu-framework-reviewers @vmware-tanzu/tkg-iam-owners
```
I would expect basically every team would eventually add themselves to this line if we go this route. I'm happy to change it back to this if its your preference!

**Comment by benjaminapetersen** [comment]
@vuil reverted to the original.

**Comment by vuil** [comment]
>are there docs folks that actually oversee these files? maybe a core reviewer could answer that question when they review this pr.

Good question. We should but the logistics is still being worked out. For now I created a team tanzu-framework-docs that we should use for mapping this PR needs, and we can sort out the membership later.

**Review by vuil — COMMENTED** [review]
@benjaminapetersen can you make the update to include other review teams as well? 
I also responded to @ankeesler's comment and created a placeholder tanzu-framework-docs team to help with the review in the future as well.

So in short I suggest we map /docs to

```@vmware-tanzu/tanzu-framework-reviewers @vmware-tanzu/tkg-iam-owners @vmware-tanzu/tkg-addons-owners @vmware-tanzu/tanzu-framework-lcm-reviewers @vmware-tanzu/tanzu-framework-docs```

**Comment by vuil** [comment]
I took the liberty to update the file as suggested, and in addition created another mapping entry for UI-related changes to be assigned to vmware-tanzu/tanzu-framework-ui-reviewers

**Comment by vuil** [comment]
Also incorporated suggestions on : https://github.com/vmware-tanzu/tanzu-framework/pull/1194
@jayunit100

**Review by anujc25 — APPROVED** [review]
lgtm

**Pull request merged**